test/smartbuffer.test.ts

```diff
@@ -614,26 +614,44 @@ describe('Reading/Writing To/From SmartBuffer', () => {
 });
 
 describe('Skipping around data', () => {
```

Comment:
Just to explain the diff noise in this file:

There was already this `Skipping around data` test block that had tests manipulating the `readOffset` to skip. I replaced this block's tests with ones for `readSkip`/`writeSkip`, and moved the old tests* to the `Setting write and read offsets` block, since that felt like the more appropriate place for them. 

\* Actually, in the case of the `Should throw an error when attempting to skip more bytes than actually exist` test, I removed it completely because you already had other tests that covered it.